bin/citrea/build.rs

```diff
@@ -12,4 +16,47 @@ fn main() {
             .args(["install", "cargo-risczero"])
             .status();
     }
+
+    build_sp1_guest();
+}
+
+fn build_sp1_guest() {
+    println!("cargo:rerun-if-env-changed=BUILD_SP1_GUEST");
+
+    let bitcoin_program_path = "provers/sp1/guest-bitcoin";
+
+    let build_args = sp1_helper::BuildArgs {
+        elf_name: "zkvm-elf".to_string(),
+        ..Default::default()
+    };
+
+    match std::env::var("BUILD_SP1_GUEST") {
```

Comment:
Shouldn't this be the default? If we want to be consistent, we should have a flag `SKIP_GUEST_BUILD` similar to the risc0 one.

Reply:
The thing is, we are not using SP1 currently in any code-paths of the citrea program. And building SP1 code adds unnecessary build time. Hence, I made the building opt-in rather than opt-out. If you wanna play around with it you have to specify it.

Reply:
Once we select one of the zkvm providers, we will still use SKIP for that ofc.